aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -146,9 +146,17 @@ std::tuple<Tensor &,Tensor &> _max_out_cpu(Tensor& max, Tensor& max_indices,
 }
 
 std::tuple<Tensor, Tensor> max(const Tensor& self, int64_t dim, bool keepdim) {
-  Tensor max = at::empty({0}, self.options());
   Tensor max_indices = at::empty({0}, self.options().dtype(kLong));
-  return at::native::max_out(max, max_indices, self, dim, keepdim);
+  if (self.is_quantized()) {
+    Tensor max = at::empty({0}, self.options().dtype(toUnderlying(self.scalar_type())));
+    Tensor max_, max_indices_;
+    at::native::max_out(max, max_indices, self.int_repr(), dim, keepdim);
+    // TODO: qscheme
+    return std::tuple<Tensor, Tensor>(at::_per_tensor_affine_qtensor(max, self.q_scale().toDouble(), self.q_zero_point().toLong()), max_indices);
```

Comment:
btw, unrelated question. Why do q_scale() return Scalar but perf_tensor_affine expect double? Shouldn't they be consistent, one way or another?

Reply:
hmm, OK, maybe per tensor affine should expect Scalar

Reply:
then should we make everything Scalar? including args of `quantize_linear`?

Reply:
why not make everything regular types? (int64/float). Do we expect ever scale to be integer or zero point to be float? (I guess zero point could theoretically, but I doubt it)